modules/@apostrophecms/schema/index.js

```diff
@@ -505,39 +505,38 @@ module.exports = {
           if (convert) {
             try {
               await convert(req, field, data, destination);
-            } catch (e) {
-              if (Array.isArray(e)) {
+            } catch (error) {
+              if ((error.name === 'required' || error.name === 'mandatory')) {
+                // `self.isVisible` will throw if an external condition key contains an unknown module
+                // or method, only for required fields
+                const isVisible = await self.isVisible(req, schema, data, field.name);
```

Comment:
Using `data` rather than `destination` because the latter won't contain all the fields, until the loop has ended.
We could do as before and check errors _after_ the loop but we'd have the errors displayed for hidden required fields... we're in a chicken and egg situation here.

We could still call the `self.apos.util.error(error + '\n\n' + error.stack);` line conditionally, after the loop, do avoid displaying an error for hidden required field, if using `data` is an issue (that from what I know isn't, but I can miss something).